packages/astro/src/manifest/create.ts

```diff
@@ -0,0 +1,283 @@
+import fs from 'fs';
+import path from 'path';
+import { compile } from 'path-to-regexp';
+import slash from 'slash';
+import { fileURLToPath } from 'url';
+import { AstroConfig, ManifestData, RouteData } from '../@types/astro';
+interface Part {
+  content: string;
+  dynamic: boolean;
+  spread: boolean;
+}
+
+interface Item {
+  basename: string;
+  ext: string;
+  parts: Part[];
+  file: string;
+  isDir: boolean;
+  isIndex: boolean;
+  isPage: boolean;
+  routeSuffix: string;
+}
+
+// Needed?
+// const specials = new Set([]);
+
+export function createManifest({ config, cwd }: { config: AstroConfig; cwd?: string }): ManifestData {
+  const components: string[] = [];
+  const routes: RouteData[] = [];
+
+  function walk(dir: string, parentSegments: Part[][], parentParams: string[]) {
+    let items: Item[] = [];
+    fs.readdirSync(dir).forEach((basename) => {
+      const resolved = path.join(dir, basename);
+      const file = slash(path.relative(cwd || fileURLToPath(config.projectRoot), resolved));
+      const isDir = fs.statSync(resolved).isDirectory();
+
+      const ext = path.extname(basename);
+      const name = ext ? basename.slice(0, -ext.length) : basename;
+
+
+			if (name[0] === '_') {
+				return;
+			}
+      if (basename[0] === '.' && basename !== '.well-known') {
+        return;
+      }
+      if (!isDir && !/^(\.[a-z0-9]+)+$/i.test(ext)) {
```

Comment:
Note: for dotfiles, `ext` will be `''` so you may not need this RegEx

Reply:
This is testing specifically for weird temp files, like `foo.astro_tmp`. This was copied from SvelteKit, bu I'm not sure how useful it is, but I guess it can't hurt? Updated comment to be more specific.